testsuite/MDAnalysisTests/auxiliary/base.py

```diff
@@ -357,8 +357,32 @@ def test_load_from_description(self, reader):
         new = mda.auxiliary.core.auxreader(**description)
         assert new == reader, "AuxReader reloaded from description does not match"
 
-    
+    def test_step_to_frame_out_of_bound(self, reader):
 
+        ts = mda.coordinates.base.Timestep(0, dt=1)
+
+        assert reader.step_to_frame(-1, ts) is None
+        assert reader.step_to_frame(reader.n_steps, ts) is None
+
+    def test_step_to_frame_no_time_diff(self, reader):
+
+        ts = mda.coordinates.base.Timestep(0, dt=1)
+        
+        for idx in range(reader.n_steps):
+
+            assert reader.step_to_frame(idx, ts) == idx
+
+    def test_step_to_frame_time_diff(self, reader):
+
+        # Timestep is 0.1 longer than auxiliary data
+        ts = mda.coordinates.base.Timestep(0, dt=1.1)
+
+        for idx in range(reader.n_steps):
+
+            frame, time_diff = reader.step_to_frame(idx, ts, return_time_diff=True)
+
+            assert frame == idx
+            np.testing.assert_almost_equal(time_diff, idx * 0.1)
```

Comment:
Good point, thanks. I hard-coded a loop over 5 frames and added a comment, so that this will work if the trajectory length is changed. Let me know if you prefer I only add a comment so that this test will likely fail if the trajectory length is increase.